### PR TITLE
Add OIDC and App information to the authetication page

### PR DESCRIPTION
**Pull request by patrys**
This improves upon the previous change by documenting all authentication details in the same place.

### CODE REVIEW

**Review by CossackDex — APPROVED** [review]

File: docs/api-usage/authentication.mdx [hunk 1]
```diff
@@ -142,3 +142,268 @@ mutation {
...
+
+```http
+POST /graphql/ HTTP/1.1
+...
+Authorization: Bearer <app-token>
+```
+
+### Dashboard extensions
+
+Saleor Apps extending the dashboard will be passed a special JSON Web Token (JWT) with the effective permission set equal to the intersection of user and application permissions. This token allows your frontend part to access the API on behalf of the user.
```
Comment (lkostrowski) [inline docs/api-usage/authentication.mdx:407]: you can link https://github.com/saleor/saleor-app-sdk/blob/main/docs/app-bridge.md

**Review by lkostrowski — APPROVED** [review]

File: docs/api-usage/authentication.mdx [hunk 2]
```diff
@@ -142,3 +142,268 @@ mutation {
...
+
+If **Use OAuth scope permissions** is enabled, any permissions granted to a user on the authentication provider side will be used as the effective permissions on the Saleor side.
+
+If the plugin is configured to manage user permissions, you need to create OAuth scopes corresponding to Saleor permissions on the identity provider side. The plugin will request lowercase scope names based on the [`PermissionEnum`](api-reference/enums/permission-enum.mdx). For example, the `MANAGE_PRODUCTS` corresponds to the `saleor:manage_products` scope.
+
+Saleor will look for supported permissions in the `scope` field of the access token. If the `scope` doesn't contain any Saleor permissions, Saleor will look for permissions in the token's `permissions` field.
+
+Please see the documentation of your authentication server to see how to manage permissions and configure role-based access control (RBAC).
+
+- [Auth0](https://auth0.com/docs/manage-users/access-control/rbac)
```
[the remark this answers is not in the document]
Comment (patrys) [reply docs/api-usage/authentication.mdx:177]: Auth0 is just one vendor, I want to expand this list with docs for other platforms.

[the remark this answers is not in the document]
Comment (lkostrowski) [reply docs/api-usage/authentication.mdx:177]: OK, list with 1 item just looked like a mistake

File: docs/api-usage/authentication.mdx (hunk 1 above)
Comment (patrys) [reply docs/api-usage/authentication.mdx:407]: Please either add the link or submit a PR to add it.

BTW, as a user, I’d expect the SDK to be something covered by the docs, not an external resource.

**Review by fowczarek — APPROVED** [review]

**Pull request merged**